### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rsommerfeld**
- [x] Give `TCUTelegram` white background
- [x] Remove `NullMessage`
- [x] Rename `AuthHandler` -> `AuthenticationHandler`
- [x] Rename `TeleClient` -> `TdlibClient`
- [x] Fix switching Chat bug
- [x] Handle callbackIds in core only
- [x] Move chats attribute from core to chatsHandler
- [x] Scoll to newest Message when previously viewing the newest message
- [x] Fix chat position of newly created chats

### CODE REVIEW

File: packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/addAtBottom..st [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 	(self isFromCurrentChat: aMessage) ifFalse: [^ self].
+	viewNewestMessage := self vScrollBar maximumValue = self vScrollBar value.
 	
 	wrapper := self wrapperFor: (TCUMessage newFrom: aMessage).
 	self scroller addMorphBack: wrapper.
-	self items addLast: wrapper.
\ No newline at end of file
+	self items addLast: wrapper.
+	
+	viewNewestMessage ifTrue: [ self scrollToNewestMessage. ].
```
Comment (rsommerfeld) [inline packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/addAtBottom..st]: How about renaming to "wasViewingNewestMessage"? Could also be a separate query method

File: packages/TelegramClientTests-Core.package/TCTCAPITests.class/instance/testExtraAPIAttribute.st [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
-	| callbackID event |
-	callbackID := 'this is a callback'.
+	| callbackId event |
+	callbackId := 'this is a callback'.
 	event := nil.
-	self sharedCore pendingRequests at: callbackID put: [:anEvent | event := anEvent].
-	self sharedCore send: (TCCRequest newGetMeWith: callbackID).
+	self sharedCore pendingRequests at: callbackId put: [:anEvent | event := anEvent].
+	self sharedCore send: (TCCRequest newGetMe at: '@extra' put: callbackId; yourself).
```
Comment (rsommerfeld) [inline packages/TelegramClientTests-Core.package/TCTCAPITests.class/instance/testExtraAPIAttribute.st:8]: Maybe put cascade on next lines?

File: packages/TelegramClient-Core.package/TCCBasicGroupChat.class/instance/loadInfo.st [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-			memberCount: (aBasicGroupFullInfoEvent at: 'members') size;
-			description: (aBasicGroupFullInfoEvent at: 'description');
-			addMembersFrom: (aBasicGroupFullInfoEvent at: 'members').
-		].
-	self core send: (TCCRequest newGetBasicGroupFullInfoOf: self groupId with: callbackId).
\ No newline at end of file
+	self core
+		send: (TCCRequest newGetBasicGroupFullInfoOf: self groupId with: 'this should work')
+		thenDo: [:aBasicGroupFullInfoEvent |
+			self
```
Comment (rsommerfeld) [inline packages/TelegramClient-Core.package/TCCBasicGroupChat.class/instance/loadInfo.st:7]: Given that this is not a test, why is the text `'this should work'` there? :D

File: packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/addAtBottom..st (hunk 1 above)
Comment (jb3rndt) [reply packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/addAtBottom..st]: Good idea 👍

File: packages/TelegramClient-Core.package/TCCBasicGroupChat.class/instance/loadInfo.st (hunk 3 above)
Comment (jb3rndt) [reply packages/TelegramClient-Core.package/TCCBasicGroupChat.class/instance/loadInfo.st]: Leftovers from testing :D

File: packages/TelegramClient-Core.package/TCCChat.class/instance/lastMessageId.st [hunk 4]
```diff
@@ -0,0 +1,6 @@
+accessing
+lastMessageId
+
+	^ self messages
+		ifEmpty: [ self class defaultFromMessageId ]
```
Comment (permler) [inline packages/TelegramClient-Core.package/TCCChat.class/instance/lastMessageId.st:5]: defaultFromMessageId is a bit of a strange name, I think it's fine but maybe something like defaultLastMessageId would fit better?

File: packages/TelegramClient-Core.package/TCCRequest.class/class/newGetBasicGroupFullInfoOf..st [hunk 5]
```diff
@@ -1,9 +1,8 @@
 instance creation
-newGetBasicGroupFullInfoOf: aGroupId with: anExtraString
+newGetBasicGroupFullInfoOf: aGroupId
 
 	^ TCCRequest
 		newWithType: 'getBasicGroupFullInfo'
 		from: {
 			'basic_group_id' -> aGroupId.
-			'@extra'-> anExtraString
```
Comment (permler) [inline packages/TelegramClient-Core.package/TCCRequest.class/class/newGetBasicGroupFullInfoOf..st:8]: Why can we leave out the @extra attribute?

File: packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st [hunk 6]
```diff
@@ -0,0 +1,4 @@
+messages
+isViewingNewestMessage
+
+	^ self vScrollBar maximumValue = self vScrollBar value
```
Comment (permler) [inline packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st:4]: Are you sure this works? I believe this only tests if the newest message was fully visible. Is that what you wanted to test?

**Review by permler — COMMENTED** [review]
Great refactoring! Very nice changes :)

File: packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st (hunk 6 above)
Comment (jb3rndt) [reply packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st:4]: Yeah the intention was to have a boolean to see whether the user would like the chat to automatically scroll to every new message. From my intention this is only the case if the chat is fully scrolled down and as soon as the user scrolls up a little bit he might search for something at that place.
I am not sure if the name states the intention correctly... Do you think the method name is ok or do you think something like `fullyScrolledDown` would be better?

Comment (permler) [reply packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st:4]: I personally like `fullyScrolledDown` better :)

Comment (permler) [reply packages/TelegramClient-UI.package/TCUChatMessageList.class/instance/isViewingNewestMessage.st]: maybe isFullyScrolledDown?

**Pull request merged**